File: backend/app/domain/schemas/obra_social.py
from pydantic import BaseModel, ConfigDict
from app.domain.models import TipoObraSocialEnum

class ObraSocialBase(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    nombre: str
    cuit: str
    porcentaje_cobertura: float
    nombre_tipo: TipoObraSocialEnum

class ObraSocialCreate(ObraSocialBase):
    pass

class ObraSocialUpdate(ObraSocialBase):
    pass

class ObraSocialRead(ObraSocialBase):
    id: int

File: frontend/turnero-medico/src/types/ObraSocial.ts
export type TipoObraSocialEnum = 'Nacional' | 'Provincial' | 'Jubilado';

export interface ObraSocial {
  id: number;
  cuit: string;
  nombre: string;
  porcentaje_cobertura: number;
  nombre_tipo: TipoObraSocialEnum;
}

export interface ObraSocialCreate {
  cuit: string;
  nombre: string;
  porcentaje_cobertura: number;
  nombre_tipo: TipoObraSocialEnum;
}
export interface ObraSocialUpdate {
  cuit?: string;
  nombre?: string;
  porcentaje_cobertura?: number;
  nombre_tipo?: TipoObraSocialEnum;
}

export type ObraSocialPayload = ObraSocialCreate | ObraSocialUpdate;
